wagtail_newsletter/actions.py

```diff
@@ -0,0 +1,31 @@
+from typing import cast
+
+from wagtail.admin import messages
+
+from . import campaign_backends
+from .models import NewsletterPageMixin
+
+
+def save_campaign(request, page: NewsletterPageMixin) -> None:
```

Comment:
Well if this doesn't belong to the "business logic" layer, and is only used in the `wagtail_hooks` module...  
Shouldn't it be a private function of that module then, rather than increasing the public API of that package with a `action` module that seem to actually be private?

But I appreciate that this is a bit subjective, so feel free to merge this code as is if you think that this `action` module still has a good reason to exist  :slightly_smiling_face:

Reply:
Hmm, I'm not sure I agree with this definition of what is part of the public API and what isn't. Rather I think that whatever is referenced in the documentation is part of the library's contract (and is thus "public"); everything else is an implementation detail. I'm not particularly eager to add underscores to functions that get called across modules (though functions that are internal to a module seem like good candidates for underscore prefixing).

Happy to discuss this further!

Reply:
I'll merge the PR to unblock follow-on work but we can revisit the issue any time.